packages/venia-ui/lib/actions/user/asyncActions.js

```diff
@@ -79,61 +79,44 @@ export const getUserDetails = () =>
         }
     };
 
-export const createNewUserRequest = accountInfo =>
-    async function thunk(...args) {
-        const [dispatch] = args;
-
-        dispatch(actions.resetCreateAccountError.request());
-
-        try {
-            await request('/rest/V1/customers', {
-                method: 'POST',
-                body: JSON.stringify(accountInfo)
-            });
-
-            await dispatch(
-                signIn({
-                    username: accountInfo.customer.email,
-                    password: accountInfo.password
-                })
-            );
-        } catch (error) {
-            dispatch(actions.createAccountError.receive(error));
-
-            /*
-             * Throw error again to notify async action which dispatched handleCreateAccount.
-             */
-            throw error;
-        }
-    };
-
 export const createAccount = accountInfo => async dispatch => {
-    /*
-     * Server validation error is handled in handleCreateAccount.
-     * We set createAccountError in Redux and throw error again
-     * to notify redux-thunk action which dispatched handleCreateAccount action.
-     */
+    dispatch(actions.createAccount.request());
+
     try {
-        await dispatch(createNewUserRequest(accountInfo));
-    } catch (e) {}
+        await request('/rest/V1/customers', {
+            method: 'POST',
+            body: JSON.stringify(accountInfo)
+        });
+
+        await dispatch(
+            signIn({
+                username: accountInfo.customer.email,
+                password: accountInfo.password
+            })
+        );
+    } catch (error) {
+        dispatch(actions.createAccount.receive(error));
+
+        /*
+         * Throw error again to notify async action which dispatched handleCreateAccount.
+         */
+        throw error;
+    }
 };
 
 export const resetPassword = ({ email }) =>
     async function thunk(...args) {
         const [dispatch] = args;
 
-        dispatch(actions.resetPassword.request(email));
+        dispatch(actions.resetPassword.request());
 
         // TODO: actually make the call to the API.
         // For now, just return a resolved promise.
-        const response = await Promise.resolve(email);
+        await Promise.resolve(email);
 
-        dispatch(actions.resetPassword.receive(response));
+        dispatch(actions.resetPassword.receive());
```

Comment:
Yes. I removed payloads everywhere they were unused. In this case I refactored how the reset password feature worked so that I didn't have to maintain the submitted password in global state.